crates/compiler/gen_llvm/src/llvm/bitcode.rs

```diff
@@ -243,7 +243,7 @@ fn build_transform_caller_help<'a, 'ctx>(
     let block = env.builder.get_insert_block().expect("to be in a function");
     let di_location = env.builder.get_current_debug_location().unwrap();
 
-    let arg_type = env.context.i8_type().ptr_type(AddressSpace::default());
+    let arg_type = env.context.ptr_type(AddressSpace::default());
```

Comment:
Since LLVM 15 pointers are not typed. Using the latest inkwell surfaces this as a warning. The fix is just to replace the ptr types with `env.context.ptr_type(AddressSpace::default())` as advised by cargo check. 

After making this change, there were a number of places where variables were no longer used, so I followed chippy's suggestions and removed each of these.